### PR TITLE
Multiple transport protocols

### LINKED ISSUE #16: [Handshake] Allow different transport protocols
RE: BOTD#2

TL;DR (bu read anyhow) we need to support multiple transport, not a single one.

Long explanation:
for browsers and mobile app to consume the daemon rpcs would be to necessary to have also server-side TLS, otherwise would be marked as insecure by browser and not allowed in app stores by mobile apps. So this means that if we support Noise-like protocols, we should mandatory support also TLS. I guess this is doable, but this led to the question if we should impose a strict secure transport over another. Noise + TLS would also make it slower (ie double encryption), without considering instead that a lot of peers would want to use tor hidden services too.

My proposal:

- rename Handshake protocol to Transport protocol
- each peer should advertise (via cleartext HTTP1 endpoint) which set of protocols supports
- It should be a one (or couple) between: insecure, onion, TLS, mTLS, Noise

### PR DESCRIPTION
**Pull request by tiero**
Before this commit the the BOTD#2 spec was describing an opinionated Handshake protocol unsing ECDH and AES256 crypto, very similar to Noise protocol. 

Now  in the spec we allow each peer to chose how to secure the transport (with or without handshake) having a list to choose one or more of them.

It  renames the BOTD#2 from handshake to transport protocol.

It closes #16

### CODE REVIEW

**Review by CL-Seven — APPROVED** [review]

File: 01-message-protocol.md [hunk 1]
```diff
@@ -4,24 +4,8 @@
...
-syntax = "proto3";
-
-message SecretMessage {
-  // Number of bytes in payload
-  uint32 lenght = 1;
-  // The encrypted Protobuf message serialized as bytes
-  bytes payload = 2;
-}
-```
+The messages exchnaged between peers are serialized using [**Protocol Buffers**](https://developers.google.com/protocol-buffers/), a language-neutral, platform-neutral, extensible mechanism for serializing structured data.
```
Comment (altafan) [inline 01-message-protocol.md:9]: typo:
```suggestion
The messages exchanged between peers are serialized using [**Protocol Buffers**](https://developers.google.com/protocol-buffers/), a language-neutral, platform-neutral, extensible mechanism for serializing structured data.
```

File: 02-transport-protocol.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+## Overview
+
+All communications between two *peers* SHOULD be encrypted in order to provide confidentiality for all transcripts and is authenticated in order to avoid malicious interference. 
+
+
+
+### Transport announcement
+
+Each *peer* MUST decides and MUST anounces which transport protocol wants to use for the connection, one or more combined between:
```
Comment (altafan) [inline 02-transport-protocol.md:12]:
```suggestion
Each *peer* MUST decide and MUST announce which transport protocol wants to use for the connection, one or more combined between:
```

File: 02-transport-protocol.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+
+* Clear text `insecure`
+* Onion service `onion`
+* Server-side TLS `TLS`
+* Mutual TLS `mTLS`
+* Noise_XK_secp256k1_ChaChaPoly_SHA256 `noise`
+
+
+Each peer SHOULD be reachable on the default TCP port **9945** and, if so, MUST expose then a plaintext HTTP/2 endpoint that announces the chosen transports responding with an `AvailableTransport` protobuf message, setting the response as `ContentType` as `application/protobuf`
```
Comment (altafan) [inline 02-transport-protocol.md:22]:
```suggestion
Each peer SHOULD be reachable on the default TCP port **9945** and, if so, MUST expose then a plaintext HTTP/2 endpoint that announces the chosen transports responding with an `AvailableTransport` protobuf message, setting the response `ContentType` as `application/protobuf`
```

>setting the response `ContentType` as `application/protobuf`

What are you trying to say with this? Why are you talking of ContentType that reminds to HTTP/1 request/response headers?

**Review by altafan — CHANGES_REQUESTED** [review]

File: 02-transport-protocol.md (hunk 3 above)
Comment (tiero) [reply 02-transport-protocol.md:22]: Well HTTP/2 has content type same as 1 :D What I'm trying to say is that we are not forcing here to be gRPC, but plain http/2 with protobuf message as response.

**Pull request merged**